pkg/cmd/infra/router/router.go

```diff
@@ -27,28 +28,104 @@ This command launches a router connected to your cluster master. The router list
 created by users and keeps a local router configuration up to date with those changes.`
 )
 
-type templateRouterConfig struct {
-	Config             *clientcmd.Config
-	TemplateFile       string
-	ReloadScript       string
+// routerConfig represents a container for all other router config types.
+// Any global config belongs on this level.
+type routerConfig struct {
```

Comment:
I'll hold off on the refactoring effort here until I understand how you mean for the flag bindings to be handled.  Is this refactoring perhaps deferrable?